File: android/app/src/main/java/app/releaf/mobile/features/notepad/recents/ui/RecentsScreen.kt
package app.releaf.mobile.features.notepad.recents.ui

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.text.BasicText
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import app.releaf.mobile.features.notepad.recents.data.MockData
import app.releaf.mobile.features.notepad.recents.model.RecentsDayStats
import app.releaf.mobile.features.notepad.recents.model.RecentsPage
import app.releaf.mobile.features.notepad.recents.model.isImported
import app.releaf.mobile.features.notepad.recents.theme.AccentImport
import app.releaf.mobile.features.notepad.recents.theme.BgCanvas
import app.releaf.mobile.features.notepad.recents.theme.TextGreenMuted
import app.releaf.mobile.features.notepad.recents.theme.TextMuted
import app.releaf.mobile.features.notepad.recents.theme.Type
import app.releaf.mobile.ui.components.CaptureMode
import java.time.format.DateTimeFormatter

/**
 * The Recents tab. The host (NotepadScreen) renders the brand
 * eyebrow, H1, Day/Recents toggle, and category filter row above this
 * branch, so this screen starts straight at the stats strip and never
 * repeats them. The local tag-chip row was removed for the same reason
 * — the host's category filter sits in the same visual position. The
 * inline 5-cell bottom nav was also removed; the host app's main
 * floating footer continues to handle navigation. It composes, in
 * order:
 *
 *   1. StatsStrip
 *   2. "TODAY" + TodayHero
 *   3. "THIS WEEK" + WeekPulse
 *   4. "EARLIER IN <MONTH>" + EarlierGrid
 *
 * Stateless externally — pass in callbacks for content actions.
 */
@Composable
fun RecentsScreen(
    stats: RecentsDayStats = MockData.dayStats,
    onOpenPage: (RecentsPage) -> Unit = {},
    /// Fires when the user picks a type from the new-entry slot's
    /// 5-cell grid (Photo / Scan / Voice / Todo / Contact) or taps
    /// the new-entry slot's footer CTA. The host is expected to open
    /// the page editor; once the editor learns to focus a specific
    /// tab, the [CaptureMode] forwarded here is the right value.
    onPickMode: (CaptureMode) -> Unit = {},
) {
    // Host (NotepadScreen) already wraps this branch in a Column with
    // verticalScroll, so RecentsScreen is a plain inline column — no
    // self-scroll, no fillMaxSize. Wrapping its own scroll inside the
    // host's would crash with "Vertically scrollable component was
    // measured with an infinity maximum height constraints".
    //
    // The host also already renders the brand eyebrow, H1, Day/Recents
    // toggle, and category filter row above this branch, so this screen
    // starts straight at the stats strip and never repeats them.
    Column(
        modifier = Modifier
            .fillMaxWidth()
            .background(BgCanvas),
    ) {
        Spacer(Modifier.height(4.dp))

        // 1. Stats strip
        val monthLabel = stats.today?.date
            ?.format(DateTimeFormatter.ofPattern("MMM"))
            ?: stats.weekPulse.lastOrNull()?.date?.format(DateTimeFormatter.ofPattern("MMM"))
            ?: "—"
        // No inner horizontal inset — the strip spans the full width
        // of the recents column (only the host screen's outer padding
        // shows on the sides), wider than the today hero below it.
        StatsStrip(
            totals = stats.totals,
            monthLabel = monthLabel,
        )
        Spacer(Modifier.height(18.dp))

        // 2. TODAY
        SectionLabel(
            text = "TODAY",
            modifier = Modifier.padding(horizontal = 14.dp),
        )
        Spacer(Modifier.height(8.dp))
        stats.today?.let { day ->
            TodayHero(
                day = day,
                onOpenPage = onOpenPage,
                onPickMode = onPickMode,
                modifier = Modifier.padding(horizontal = 6.dp),
            )
        }
        Spacer(Modifier.height(20.dp))

        // 3. THIS WEEK
        SectionLabel(
            text = "THIS WEEK",
            modifier = Modifier.padding(horizontal = 14.dp),
        )
        Spacer(Modifier.height(10.dp))
        WeekPulse(days = stats.weekPulse)
        Spacer(Modifier.height(22.dp))

        // 4. EARLIER IN <MONTH> — page-wise feed sorted by last
        // modified, freshest first. The grid no longer surfaces
        // empty days; days with no captures simply contribute zero
        // cards.
        val earlierMonth = stats.earlier.firstOrNull()
            ?.date?.format(DateTimeFormatter.ofPattern("MMMM"))?.uppercase()
            ?: monthLabel.uppercase()
        SectionLabel(
            text = "EARLIER IN $earlierMonth",
            modifier = Modifier.padding(horizontal = 14.dp),
        )
        Spacer(Modifier.height(10.dp))
        val earlierPages = stats.earlier
            .flatMap { it.pages }
            .sortedByDescending { it.updatedAt }
            .take(14)
        EarlierGrid(
            pages = earlierPages,
            onOpenPage = onOpenPage,
        )

        // Footnote legend explaining the amber dot — only renders
        // when at least one visible page (today or earlier) is
        // actually imported. Hidden when there's nothing to explain
        // so the screen doesn't accumulate a permanently-visible
        // remark for the typical user with no imported pages.
        val anyImported = (
            (stats.today?.pages.orEmpty()) + earlierPages
        ).any { it.isImported() }
        if (anyImported) {
            Spacer(Modifier.height(16.dp))
            ImportedLegend(modifier = Modifier.padding(horizontal = 14.dp))
        }
        Spacer(Modifier.height(20.dp))
    }
}

/**
 * Small footnote — amber dot + caption — explaining that pages
 * marked with the same dot in the EarlierGrid card footers were
 * imported from the photo library or a document scan.
 */
@Composable
private fun ImportedLegend(modifier: Modifier = Modifier) {
    Row(
        modifier = modifier,
        verticalAlignment = Alignment.CenterVertically,
        horizontalArrangement = Arrangement.spacedBy(6.dp),
    ) {
        Box(
            modifier = Modifier
                .size(6.dp)
                .clip(CircleShape)
                .background(AccentImport),
        )
        BasicText(
            text = "Imported from library or a scan",
            style = Type.Caption.copy(color = TextMuted, fontSize = 11.sp),
        )
    }
}


// ---------------------------------------------------------------------------
// Section label — used for "TODAY" / "THIS WEEK" / "EARLIER IN APRIL"
// ---------------------------------------------------------------------------

@Composable
private fun SectionLabel(text: String, modifier: Modifier = Modifier) {
    // Reference uses a small, muted-green eyebrow (microWide style).
    BasicText(
        text = text.uppercase(),
        style = Type.MicroWide.copy(color = TextGreenMuted),
        modifier = modifier,
    )
}

File: apps/releaf/ios/Releaf/Features/Notepad/Recents/Views/RecentsScreen.swift
import SwiftUI
import ReleafDesignSystem

/// Top-level composition for the redesigned Recents screen. Designed to drop
/// into any SwiftUI host (e.g. `WindowGroup { RecentsScreen() }` or another
/// container's `body`).
public struct RecentsScreen: View {

    private let stats: RecentsDayStats
    private let onOpenPage:    (RecentsPage) -> Void
    private let onPickMode:    (CaptureMode) -> Void

    public init(
        stats: RecentsDayStats? = nil,
        onOpenPage: @escaping (RecentsPage) -> Void = { _ in },
        /// Picker cells in the new-entry slot (Photo / Scan / Voice /
        /// Todo / Contact) and the new-entry footer CTA all funnel
        /// through this callback. The host wires it to whatever
        /// "compose new" routing it has — typically opening the editor.
        /// Once the editor learns to focus a specific tab, the
        /// `CaptureMode` forwarded here is the right value.
        onPickMode: @escaping (CaptureMode) -> Void = { _ in }
    ) {
        // Default to mock data so previews and standalone hosts render
        // out of the box. Real callers pass `RecentsAdapter.fromState(...)`.
        self.stats = stats ?? MockData.dayStats
        self.onOpenPage = onOpenPage
        self.onPickMode = onPickMode
    }

    public var body: some View {
        // Host (`NotepadView`) already wraps this branch in a `ScrollView`,
        // so RecentsScreen is rendered as an inline `VStack` — no own
        // ScrollView, no overlay bottom nav. Nesting ScrollViews would
        // produce a janky UX where the inner scroll fights the outer.
        //
        // The host also already renders the eyebrow, H1, Day/Recents
        // toggle, and category filter row above this branch, so this
        // screen starts straight at the stats strip and never repeats them.
        VStack(alignment: .leading, spacing: 18) {
            // -14pt horizontal cancels the parent VStack's 14pt margin
            // entirely, so the strip spans the full recents column
            // (only the host screen's outer padding shows on the
            // sides), wider than the today hero card below it.
            StatsStrip(totals: stats.totals, monthLabel: monthLabel)
                .padding(.horizontal, -14)

            sectionLabel("TODAY")
            TodayHero(
                day: stats.today,
                onOpenPage: onOpenPage,
                onPickMode: onPickMode
            )

            sectionLabel("THIS WEEK")
            WeekPulse(days: stats.weekPulse)

            sectionLabel("EARLIER IN \(earlierMonthLabel)")
            EarlierGrid(
                pages: earlierPages,
                onOpenPage: onOpenPage
            )

            // Footnote legend explaining the amber dot — only renders
            // when at least one visible page (today or earlier) is
            // actually imported. Hidden when there's nothing to
            // explain so the screen doesn't accumulate a permanently-
            // visible remark for the typical user with no imported
            // pages.
            if anyImportedVisible {
                importedLegend
                    .padding(.top, -2)
            }
        }
        .padding(.horizontal, 14)
        .padding(.top, 16)
        .frame(maxWidth: .infinity, alignment: .leading)
        .background(Color.bgCanvas)
    }

    private var anyImportedVisible: Bool {
        let todayPages = stats.today?.pages ?? []
        return (todayPages + earlierPages).contains(where: { $0.isImported })
    }

    /// Small footnote — amber dot + caption — explaining that pages
    /// marked with the same dot in the EarlierGrid card footers were
    /// imported from the photo library or a document scan.
    private var importedLegend: some View {
        HStack(spacing: 6) {
            Circle()
                .fill(Color.accentImport)
                .frame(width: 6, height: 6)
            Text("Imported from library or a scan")
                .font(.system(size: 11, weight: .regular))
                .foregroundColor(.textMuted)
        }
    }

    private var monthLabel: String {
        let date = stats.today?.date ?? stats.weekPulse.last?.date ?? Date()
        let f = DateFormatter()
        f.locale = Locale(identifier: "en_US_POSIX")
        f.dateFormat = "MMM"
        return f.string(from: date)
    }

    private var earlierMonthLabel: String {
        let date = stats.earlier.first?.date ?? stats.today?.date ?? Date()
        let f = DateFormatter()
        f.locale = Locale(identifier: "en_US_POSIX")
        f.dateFormat = "MMMM"
        return f.string(from: date).uppercased()
    }

    /// Flattened, last-modified-desc list of past pages, capped at
    /// 14 cards. The EarlierGrid is a page feed now (one card per
    /// page), not a day grid — empty days simply contribute zero
    /// cards.
    private var earlierPages: [RecentsPage] {
        Array(
            stats.earlier
                .flatMap(\.pages)
                .sorted(by: { $0.updatedAt > $1.updatedAt })
                .prefix(14)
        )
    }

    // (Brand label, "Recent garden" H1, and Day/Recents pill toggle
    // were intentionally removed: the host already renders all three
    // above this branch, and rendering them again here produced two
    // visually-identical headers stacked on top of each other.)

    // MARK: - Section label

    private func sectionLabel(_ text: String) -> some View {
        Text(text)
            .font(Typography.microWideFont)
            .kerning(1.6)
            .foregroundColor(.textGreenMuted)
            .padding(.top, 6)
    }

    // (Tag filtering removed alongside the local TagChips row; the
    // host's category filter row above this branch already narrows
    // entries before they reach `stats`.)
}

#if DEBUG
#Preview {
    RecentsScreen()
}
#endif
